src/java/org/apache/cassandra/journal/Flusher.java

```diff
@@ -270,15 +270,15 @@ private void waitForFlushGroup(ActiveSegment<K>.Allocation alloc)
         pending.decrementAndGet();
     }
 
-    private void asyncFlushGroup(ActiveSegment<K>.Allocation alloc, Executor executor, Runnable onDurable)
+    private void asyncFlushGroup(ActiveSegment<K>.Allocation alloc, Executor executor, AsyncWriteCallback callback)
     {
         pending.incrementAndGet();
         callbackExecutor.execute(() ->
         {
             alloc.awaitFlush(journal.metrics.waitingOnFlush);
             pending.decrementAndGet();
             written.incrementAndGet();
-            executor.execute(onDurable);
+            executor.execute(callback);
```

Comment:
rather than `AsyncWriteCallback extends Runnable` it might be better to do `callback::onSuccess`

Reply:
^ see above

Reply:
Resolving as above ^